src/core/reactor.ts

```diff
@@ -61,6 +61,51 @@ export interface Call<A, R> extends Write<A>, Read<R> {
   invoke: (args: A) => R | undefined;
 }
 
+export enum CanConnectResult {
+  SUCCESS = 0,
+  SELF_LOOP,
+  DESTINATION_OCCUPIED,
+  DOWNSTREAM_WRITE_CONFLICT,
+  NOT_IN_SCOPE,
+  RT_CONNECTION_OUTSIDE_CONTAINER,
+  RT_DIRECT_FEED_THROUGH,
+  RT_CYCLE,
+  MUTATION_CAUSALITY_LOOP
+}
+
+const CanConnectResultReasonMap = new Map<CanConnectResult, string>([
```

Comment:
Should we maybe use String enums instead? https://www.typescriptlang.org/docs/handbook/enums.html